### PR TITLE
bpo-25828: Fix PyCode_Optimize for handling exceptions

### PR DESCRIPTION
**Pull request by louisom**

### CODE REVIEW

File: Python/peephole.c [hunk 1]
```diff
@@ -475,6 +475,14 @@ PyCode_Optimize(PyObject *code, PyObject* consts, PyObject *names,
     CONST_STACK_CREATE();
 
     for (i=find_op(codestr, 0) ; i<codelen ; i=nexti) {
+        /* Bail out if an exception is set
+         * bpo-25828: fold_binops_on_constants may stuck when there is a big
+         * nubmer to calculate, e.g. 2 ** (2 ** 100). It may need to handle
+         * KeyboardInterrupt or MemoryError
+         */
+        if (PyErr_Occurred())
```
Comment (serhiy-storchaka) [inline Python/peephole.c]: Calling `PyErr_Occurred()` on every instruction can significantly slow down compilation. It should be called only after executing operations that can raise an exception.

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]

File: Python/peephole.c (hunk 1 above)
Comment (rhettinger) [reply Python/peephole.c]: I think this is a waste and isn't solving a problem that any has in the real world.

**Comment by brettcannon** [comment]
To try and help move older pull requests forward, we are going through and backfilling 'awaiting' labels on pull requests that are lacking the label. Based on the current reviews, the best we can tell in an automated fashion is that a core developer requested changes to be made to this pull request.

If/when the requested changes have been made, please leave a comment that says, `I have made the requested changes; please review again`. That will trigger a bot to flag this pull request as ready for a follow-up review.

**Comment by serhiy-storchaka** [comment]
In don't see how an exception can be set here in master.

But maybe this check is still needed in 3.6.

**Pull request closed without merging**